src/jit/morph.cpp

```diff
@@ -11932,7 +11933,16 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 case GT_BLK:
                 case GT_DYN_BLK:
                 case GT_IND:
-                    subMac1 = &subIndMac1;
+                    // If this node is part of an address computation, propagate
+                    // the existing mac.
+                    //
+                    // We can't check for this exactly, but a non-null mac is a
+                    // reasonable approximation for "this node is part of an
+                    // address computation".
```

Comment:
Do you think this is still only an approximation? What are the cases where this check is not checking for "this node is part of an address computation" exactly?

Reply:
I don't know of any cases, so I'll update the comment.